lib/ql_math.qpp

```diff
@@ -75,80 +77,143 @@ const MININT = qore(new QoreBigIntNode(-LLONG_MAX - 1));
 */
 //@{
 
-//! Returns a number equal to the closest integer to the argument passed; numbers halfway between two integers are rounded away from zero
-/** @param n a number to round
-
-    @return a number equal to the closest integer to the argument passed; numbers halfway between two integers are rounded away from zero
-
-    @par Example:
-    @code{.py}
-number n = round(num);
-    @endcode
- */
-number round(number n) [flags=CONSTANT] {
-   return qore_number_private::doUnaryNR(*n, mpfr_round);
-}
-
-//! Returns a floating-point number equal to the closest integer to the argument passed; numbers halfway between two integers are rounded away from zero
+//! Returns the multiple of <b>-prec</b>-th power of 10 that is closest to <b>num</b> (numbers halfway between two such powers of 10 are rounded away from zero). (So, for non-negative <b>prec</b> just returns <b>num</b>).
+/** @param num an integer number to round
+  * @param prec precision
+  *
+  * @par Example:
+  * @code{.py}
+  *     round(-6450, -2);  // returns  -6500
+  *     round( 6450, -3);  // returns   6000
+  *     round(-6450, -4);  // returns -10000
+  *     round( 6450, -5);  // returns      0
+  * @endcode
+  */
+int round(softint num, int prec = 0) [flags=CONSTANT] {
+    return round_ceil_floor_intern(num, prec, ROUND);
+}
+
+//! Returns the multiple of <b>-prec</b>-th power of 10 that is closest to <b>num</b> (numbers halfway between two such powers of 10 are rounded away from zero).
 /** @par Platform Availability
-    @ref Qore::Option::HAVE_ROUND
-
-    @param f a number to round
-
-    @return a floating-point number equal to the closest integer to the argument passed; numbers halfway between two integers are rounded away from zero
-
-    @par Example:
-    @code{.py}
-float n = round(num);
-    @endcode
- */
-float round(softfloat f) [flags=CONSTANT] {
+  * @ref Qore::Option::HAVE_ROUND
+  * @param num a number to round
+  * @param prec precision
+  *
+  * @par Example:
+  * @code{.py}
+  *     round( 6450.36481,  3);  // returns  6450.365
+  *     round(-6450.36481,  2);  // returns -6450.36
+  *     round( 6450.36481, -2);  // returns  6500.0
+  *     round(-6450.36481, -3);  // returns -6000.0
+  *     round( 6450.36481, -4);  // returns 10000.0
+  *     round(-6450.36481, -5);  // returns     0.0
+  * @endcode
+  */
```

Comment:
please add a doc line like: `@since %Qore 0.8.13 this function accepts the \c prec argument to specify the rounding precision`